### ISSUE TITLE
Twitter Adapter always gives Invalid CRC token or json response format.

### ISSUE THREAD

**Issue by sanjaysabnani** [BotBuilderCommunity/botbuilder-community-js#173]
Hi Team,

I have tried the twitter adapter using botbuilder.
I always get the Invalid CRC token error.
The same environment variables work for autohook by twitter.
But I want to maintain  1 code base for all bots.
Please check if something has changed from twitter.

Regards,
Sanjay Sabnani

**Comment by szul** [comment]
@sanjaysabnani Have you tried the same in the repo with the instructions? Do you get the same error?

**Comment by sanjaysabnani** [comment]
@szul 
Yes, I have used the same in the repo following all instructions.
I use the exact same environment variables and it works  perfectly for autohook.

**Comment by szul** [comment]
I'll be coming back to this shortly, as it's a part of a project that I'm working on.

**Comment by fabricioloupias** [comment]
Any news on this? The same thing happens to me, using the autohook works correctly. But not from the bot

**Comment by fabrifls** [comment]
@sanjaysabnani I don't know if the same thing is happening to you, but I had problems with the CRC token that I couldn't solve.
The solution was that the environment variable path (TWITTER_WEBHOOK_URL) points to **/api/twitter/messages**

`TWITTER_WEBHOOK_URL=https://{ngrok_domain}/api/twitter/messages`

**Comment by szul** [comment]
Sorry, but I haven't worked on this project in years and would consider the code vastly outdated. A community library like this easily turns into a full time side job which none of the previous contributors could commit to and Microsoft didn't seem willing to invest in a FTE on the JavaScript side for community development (at least at the time I was building this). You'd be better off rolling your own solution and using this repo as just historic examples of implementations.

**Comment by fabrifls** [comment]
> Sorry, but I haven't worked on this project in years and would consider the code vastly outdated. A community library like this easily turns into a full time side job which none of the previous contributors could commit to and Microsoft didn't seem willing to invest in a FTE on the JavaScript side for community development (at least at the time I was building this). You'd be better off rolling your own solution and using this repo as just historic examples of implementations.

Luckily I was able to solve it, I think I expressed myself wrong in the previous message. I don't speak English very well, sometimes the translator doesn't work as you want
